hapic/doc.py

```diff
@@ -33,6 +68,63 @@ def generate_schema_ref(spec:APISpec, schema: marshmallow.Schema) -> str:
     return ref
 
 
+def field_accepted_param(type: str, param_name:str) -> bool:
+    return (
+        param_name in FIELDS_PARAMS_GENERIC_ACCEPTED
+        or (type in FIELDS_TYPE_STRING
+            and param_name in FIELDS_PARAMS_STRING_ACCEPTED)
+        or (type in FIELDS_TYPE_ARRAY
+            and param_name in FIELDS_PARAMS_ARRAY_ACCEPTED)
+        or (type in FIELDS_TYPE_NUMERIC
+            and param_name in FIELDS_PARAMS_NUMERIC_ACCEPTED)
+    )
+
+
+def generate_fields_description(
+    schema,
+    in_: str,
+    name: str,
+    required: bool,
+    type: str=None,
+) -> dict:
+    """
+    Generate field OpenApiDescription for
+    both query and path params
+    :param schema: field schema
+    :param in_: in field
+    :param name: field name
+    :param required: required field
+    :param type: type field
+    :return: File description for OpenApi
+    """
+    description = {}
+    # INFO - G.M - 01-06-2018 - get field
+    # type to know which params are accepted
+    if not type and 'type' in schema:
+        type = schema['type']
+    assert type
+
+    for param_name, value in schema.items():
+        if field_accepted_param(type, param_name):
+            description[param_name] = value
+    description['type'] = type
+    description['in'] = in_
+    description['name'] = name
+    description['required'] = required
+
+
+    # INFO - G.M - 01-06-2018 - example is not allowed in query/path params,
+    # in OpenApi2, remove it and set it as string in field description.
+    if 'example' in schema:
+        if 'description' not in description:
+            description['description'] = ""
+        description['description'] = '{description}\n\n*example value: {example}*'.format(  # nopep8
+            description=description['description'],
+            example=schema['example']
+        )
+    return description
+
+
 def bottle_generate_operations(
```

Comment:
Actually, why do we have a bottle related function in the `hapic/doc.py` module (which should include generic stuff only) ?

Reply:
legacy name of this method, Must be modify .

Reply:
see  #64